[aries-typedevent] Fix for ARIES-2195 - Avoid OOM in history configuration

timothyjward opened the pull request:
This commit adds tests contributed by Guido Grune and fixes the issue where policies with a defined minimum and large maximum could trigger an OOM. The simple fix is to limit the maximum storage using the overall maximum number of events that the history is permitted to store. This may still waste memory if there are many different policies set with large maximums and a large total history capacity, but this can be managed by users selecting appropriate policies and overall capacities.

timothyjward committed:
[aries-typedevent] Fix for ARIES-2195 - Avoid OOM in history configuration

This commit adds tests contributed by Guido Grune and fixes the issue where policies with a defined minimum and large maximum could trigger an OOM. The simple fix is to limit the maximum storage using the overall maximum number of events that the history is permitted to store. This may still waste memory if there are many different policies set with large maximums and a large total history capacity, but this can be managed by users selecting appropriate policies and overall capacities.

Signed-off-by: Tim Ward <timothyjward@apache.org>

timothyjward commented:
Raised by @gg-dim - anything else that you think might need to be checked?

gg-dim reviewed (commented):
I have some suggestions in #29

timothyjward committed:
Include recommendations from PR review

Thanks to Guido Grune for the review. I have incorporated some of the suggested changes and added more tests.

Signed-off-by: Tim Ward <timothyjward@apache.org>
